File: lab/python/tecnicas/pilha2.py
PILHA = []


def esta_vazia():
    return not PILHA


def empilha(valor):
    PILHA.append(valor)


def desempilha():
    if esta_vazia():
        return None
    return PILHA.pop()


def topo():
    return PILHA[-1]


def tamanho():
    return len(PILHA)


if __name__ == "__main__":
    empilha(10)
    empilha(20)
    empilha(30)
    print(desempilha())
    print(topo())
    print(PILHA)
    print(tamanho())

File: lab/javascript/tecnicas/pilha2.js
// Implementação de pilha usando array
const PILHA = [];

function estaVazia() {
  return PILHA.length === 0;
}

function empilha(valor) {
  PILHA.push(valor);
}

function desempilha() {
  if (estaVazia()) {
    return null; // mesmo comportamento do Python (None)
  }
  return PILHA.pop();
}

function topo() {
  if (estaVazia()) {
    return null;
  }
  return PILHA[PILHA.length - 1];
}

function tamanho(){
    return PILHA.length
}

// --- Teste equivalente ao Python ---
empilha(10);
empilha(20);
empilha(30);

console.log(desempilha()); // 30
console.log(topo());       // 20
console.log(PILHA);        // [10, 20]
console.log(tamanho())
